lib/record_loader/application_record_loader.rb

```diff
@@ -9,5 +10,21 @@ class ApplicationRecordLoader < RecordLoader::Base
     # Uses the standard RailsAdapter
     # @see https://rubydoc.info/github/sanger/record_loader/RecordLoader/Adapter
     adapter RecordLoader::Adapter::Rails.new
+
+    def wip_list
+      # return a list of WIP files name as features if deploy_wip_pipelines is set to true, or return empty list
+      deploy_wip_pipelines = Rails.application.config.try(:deploy_wip_pipelines) || false
+      return [] unless deploy_wip_pipelines
+
+      wip_files = []
+      wip_files_path = @path
```

Comment:
Where does this `@path` instance variable come from?

Reply:
I see, it comes from the `RecordLoader::Base` class within the Record Loader gem - https://github.com/sanger/record_loader/blob/master/lib/record_loader/base.rb

This feels like bad practice, because that instance variable is not explicitly exposed / documented so could change in a future version of the gem. But since it's our gem and the alternative is to hardcode the path `config/default_records`, it's probably fine for now! I guess to make it better we'd have to modify the gem itself. 

Can you add a comment above this line explaining where `@path` comes from? Thanks.

Reply:
I would like to record a few notes here.

The "@path" field is initialised from the directory argument passed to the initialiser method of the super class. The initialiser method is inherited just like any other method in ruby. Using the stored "@path" in the child allows testability by passing a different directory in tests. 

The comment says, "@path returns 'config/default_records' directory", which is not quite true. It is whatever the directory argument passed to the initialiser (after a few checks), but it defaults to 'config/default_records' .

As the superclass does not provide a getter or an attribute reader for "@path", the subclass accessed it directly in the subclass. There is no strict private fields in ruby either. It is a bad practice but as Katy said alternative would need modifying the superclass as well.